src/app/server/Mdns.cpp

```diff
@@ -78,37 +78,204 @@ chip::ByteSpan FillMAC(uint8_t (&mac)[8])
 
 } // namespace
 
-uint16_t gSecuredPort   = CHIP_PORT;
-uint16_t gUnsecuredPort = CHIP_UDC_PORT;
+#if CHIP_DEVICE_CONFIG_ENABLE_EXTENDED_DISCOVERY
+
+constexpr const char kExtendedDiscoveryTimeoutKeypairStorage[] = "ExtDiscKey";
+
+void MdnsServer::SetExtendedDiscoveryTimeoutSecs(int16_t secs)
+{
+    ChipLogDetail(Discovery, "SetExtendedDiscoveryTimeoutSecs %d", secs);
+    chip::DeviceLayer::PersistedStorage::KeyValueStoreMgr().Put(kExtendedDiscoveryTimeoutKeypairStorage, &secs, sizeof(secs));
+}
+
+int16_t MdnsServer::GetExtendedDiscoveryTimeoutSecs()
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    int16_t secs;
+
+    err = chip::DeviceLayer::PersistedStorage::KeyValueStoreMgr().Get(kExtendedDiscoveryTimeoutKeypairStorage, &secs, sizeof(secs));
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Discovery, "Failed to get extended timeout configuration err: %s", chip::ErrorStr(err));
+        secs = CHIP_DEVICE_CONFIG_EXTENDED_DISCOVERY_TIMEOUT_SECS;
+    }
+
+    ChipLogDetail(Discovery, "GetExtendedDiscoveryTimeoutSecs %d", secs);
+    return secs;
+}
+
+/// Callback from Extended Discovery Expiration timer
```

Comment:
doccomments should be in the header not cpp file (when people see the interface they use, they read the header)

Reply:
this method is not in the header, but for the others I moved the comments